test/parallel/test-net-server-no-handler.js

```diff
@@ -0,0 +1,12 @@
+// Ensure that a net.Server without 'connection' event listeners closes
+// incoming connections rather than accepting, then forgetting about them.
+
+'use strict';
+const common = require('../common');
+const net = require('net');
+
+net.createServer(common.mustCall()).listen(function() {
+  const server = this;
+  const { address: host, port } = server.address();
+  net.connect(port, host).once('end', () => server.close());
```

Comment:
should probably be `once('close', ...`

Reply:
@bnoordhuis: I think if you change this to `'close'` we can land this PR.

Reply:
@ronag At least locally for me, that doesn’t work :confused:

Reply:
I'll take a look this week.

Reply:
@addaleax This seems to be related to a bug on the tcp handle. _handle.readStart is called but _handle.onread is never invoked, in this case I guess it should be an error such as ECONNRESET.

Reply:
This is probably a bug that also affects the `self.maxConnections` guard.

Reply:
@ronag I feel like we had a conversation about this recently, assuming that we’re talking about the client handle here – namely that `.close()` doesn’t necessarily actually send a RST? I guess that means that we should shutdown the handle here (or `.end()` the created stream, which is a bit more expensive but avoids having to manually call `.shutdown()`)?

Reply:
> namely that .close() doesn’t necessarily actually send a RST

I thought this wasn't necessary. That's at least what I got from the conversation and that the handle should detect ungraceful disconnection?

Reply:
https://github.com/nodejs/node/issues/31916